ibis/expr/types/relations.py

```diff
@@ -696,13 +697,19 @@ def drop(self, fields: str | Sequence[str]) -> Table:
         Returns
         -------
         Table
-            Expression without `fields`
+            A table with all columns in `fields` removed.
         """
         if not fields:
             # no-op if nothing to be dropped
             return self
 
-        fields = util.promote_list(fields)
+        if len(fields) == 1 and not isinstance(fields[0], str):
+            fields = util.promote_list(fields[0])
+            warnings.warn(
+                "Passing a sequence of fields to `drop` is deprecated, "
+                "use `drop(*fields)` instead",
```

Comment:
Shall we mark a version here?

Reply:
Happy to if we know when the deprecation cycle should complete. I'm not sure what our deprecation policies are.

Reply:
I think removing it in 5.0 would be fine.

Reply:
Done.